### PR TITLE
Document that uv_read_stop always succeeds

### LINKED ISSUE #3041: Add note to the documentation about checking uv_read_stop return value
* **Version**: master branch
* **Platform**: all platforms

uv_read_stop has an int return type and yet most code never checks the return value.

The answer is that, in most cases, the return value is always going to be zero so there is no need to check.

The exception is for uv_tty_t on Windows only, where uv_read_stop may return an error. On other platforms it always returns zero, and the same for everything other than uv_tty_t on Windows. (I guess there is always the possibility that in the future, some new platform or new feature may be added in which uv_read_stop can return non-zero in other cases.)

I think it is true though that even if uv_read_stop returns an error on a Windows tty_t, it still will stop the read callback from being called?

This topic confused me, I found the answer reading the code, but I suggest the documentation should be extended, so other people don't have to read the code to find the answer.

### PR DESCRIPTION
**Pull request by skissane**
Addresses #3041

### CODE REVIEW

**Comment by skissane** [comment]
I will be honest I'm not a big fan of this "stale bot"

**Comment by skissane** [comment]
Not stale

**Comment by skissane** [comment]
not stale

File: docs/src/stream.rst [hunk 1]
```diff
@@ -145,6 +145,9 @@ API
     no longer be called.
 
     This function is idempotent and may be safely called on a stopped stream.
+    
+    On most platforms, this function will always succeed; hence, checking its
+    return value is unnecessary.
```
Comment (vtjnash) [inline docs/src/stream.rst:150]: How about this:
```suggestion
    This function will always succeed; hence, checking its return value is
    unnecessary. A non-zero return indicates that finishing releasing resources
    may be pending on the next input event on that TTY on Windows, and does not 
    indicate failure.
```

**Comment by vtjnash** [comment]
I missed that this was against master, so I pushed it to v1.x also in 4cf638de0505035506521a5988193b9b8d94a281. It will probably cause a merge conflict when we update master, so should I force-push master to the previous commit?

**Pull request merged**